### PR TITLE
Revise changelog entries for 0.8.0 release

### PR DESCRIPTION
**Pull request by namurphy**
This PR (part of #1595) goes through the changelog entries and makes some minor edits.  This was pretty straightforward this time, since we've been doing a really good job looking at changelog entries during code review.

This PR complements #1587, which updates some reST links (which mostly pointed to objects contained in `plasmapy.formulary.parameters`.

### CODE REVIEW

File: changelog/974.breaking.rst [hunk 1]
```diff
@@ -1,2 +1,2 @@
 In `~plasmapy.diagnostics.thomson.spectral_density`, the arguments``Te`` and ``Ti`` have been renamed ``T_e`` and ``T_i``
```
Comment (rocco8773) [inline changelog/974.breaking.rst:1]:
```suggestion
In `~plasmapy.diagnostics.thomson.spectral_density`, the arguments ``Te`` and ``Ti`` have been renamed ``T_e`` and ``T_i``
```

File: changelog/1591.trivial.rst [hunk 2]
```diff
@@ -1 +0,0 @@
-Fixed warnings in plasmapy/particles/particle_class.py (deprecated escape sequence).
```
Comment (rocco8773) [inline changelog/1591.trivial.rst:1]: Why did you remove the changelog entry?

Comment (namurphy) [reply changelog/1591.trivial.rst:1]: This PR was changing a `"""` to `r"""` for a docstring, so we can get away without one.
